palamedes/__init__.py

```diff
@@ -1,16 +1,152 @@
-from Bio.Align import PairwiseAligner
+from Bio.Align import PairwiseAligner, Alignment
 from Bio.SeqRecord import SeqRecord
 from hgvs.sequencevariant import SequenceVariant
 
-from palamedes.config import MOLECULE_TYPE_PROTEIN, ALT_SEQUENCE_ID, REF_SEQUENCE_ID
-from palamedes.align import generate_seq_record, generate_alignment, generate_variant_blocks
+from palamedes.align import generate_seq_record, generate_variant_blocks, reverse_seq_record
 from palamedes.hgvs.utils import categorize_variant_block
 from palamedes.hgvs.builders import BUILDER_CONFIG
-
+from palamedes.config import (
+    DEFAULT_EXTEND_GAP_SCORE,
+    DEFAULT_MATCH_SCORE,
+    DEFAULT_MISMATCH_SCORE,
+    DEFAULT_OPEN_GAP_SCORE,
+    GLOBAL_ALIGN_MODE,
+    MOLECULE_TYPE_PROTEIN,
+     ALT_SEQUENCE_ID,
+     REF_SEQUENCE_ID,
+)
 
 __version__ = "0.0.8"
 
 
+def generate_hgvs_variants_from_alignment(
+    alignment: Alignment, use_non_standard_substitution_rules: bool = False, molecule_type: str = MOLECULE_TYPE_PROTEIN
+) -> list[SequenceVariant]:
+    """
+    Given a pairwise alignment object and a molecule type, generate a list of HGVS SequenceVariants.
+
+    - Alignment: Generated via BioPython.PairwiseAligner - (`generate_alignment` for more information)
+
+    - molecule_type: Currently only molecule type 'protein' is supported.
+
+    - An optional flag: `use_non_standard_substitution_rules` is a boolean flag which will enable logic that treats
+    multiple consecutive mismatches as separate subsitutions, vs merging together into a delins. This is against HGVS
+    spec but has utility for some use cases.
+
+
+    """
+    if molecule_type not in BUILDER_CONFIG:
+        raise NotImplementedError(f"No HGVS builder is defined for molecule_type: {molecule_type}!")
+
+    variant_blocks = generate_variant_blocks(
+        alignment,
+        split_consecutive_mismatches=use_non_standard_substitution_rules,
+    )
+    builder = BUILDER_CONFIG[molecule_type](alignment)
+    return [
+        builder.build(
+            variant_block,
+            categorize_variant_block(variant_block, alignment),
+        )
+        for variant_block in variant_blocks
+    ]
+
+
+def generate_alignment(
+    reference_seq_record: SeqRecord,
+    alternate_seq_record: SeqRecord,
+    molecule_type: str = MOLECULE_TYPE_PROTEIN,
+    aligner: PairwiseAligner | None = None,
+) -> Alignment:
+    """
```

Comment:
Since this is public now, can you add some code example to the doc string, using the same block as above?